tests/foreman/sys/test_pulp3_filesystem.py

```diff
@@ -0,0 +1,89 @@
+"""Tests for Pulp3 file system.
+
+:Requirement: Pulp3 installed
+
+:CaseAutomation: Automated
+
+:CaseLevel: System
+
+:CaseComponent: Pulp
+
+:Assignee: ltran
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+
+"""
+import pytest
+
+from robottelo.ssh import get_connection
+
+
+@pytest.mark.upgrade
+def test_selinux_status():
+    """Test SELinux status.
+
+    :id: 43218070-ac5e-4679-b74a-3e2bcb497a0a
+
+    :expectedresults: SELinux is enabled and there are no denials
+
+    """
+    with get_connection(timeout=100) as connection:
+        # check SELinux is enabled
+        result = connection.run('getenforce')
+        assert 'Enforcing' in result.stdout[0]
+        # check there are no SELinux denials
+        result = connection.run('ausearch --input-logs -m avc -ts today --raw')
+        assert result.return_code == 1
+
+
+@pytest.mark.upgrade
+def test_pulp_directory_exists():
+    """Check Pulp3 directories are present.
+
+    :id: 96a64932-9e89-4063-9d5b-55c811375361
+
+    :expectedresults: Directories are present
+
+    """
+    with get_connection(timeout=100) as connection:
+        # check pulp3 directories present
+        directories = [
+            '/var/lib/pulp',
+            '/var/lib/pulp/assets',
+            '/var/lib/pulp/media',
+            '/var/lib/pulp/tmp',
+            '/var/lib/pulp/static',
+        ]
+        for directory in directories:
+            assert (
+                connection.run(f'test -d {directory}').return_code == 0
+            ), f'{directory} must exist.'
```

Comment:
My suggestion was to replace this with:
```python
@pytest.fixture
def connection(timeout=100):
    with get_connection(timeout=timeout) as connection:
        yield connection


@pytest.mark.upgrade 
@pytest.mark.parametrize('directory', [....])
def test_pulp_directory_exists(connection, directory):
    """Check Pulp3 directories are present.
    :id: 96a64932-9e89-4063-9d5b-55c811375361
    :expectedresults: Directories are present
    """
    assert connection.run(f'test -d {directory}').return_code == 0
```

Reply:
OK, done

Reply:
I still don't see `@pytest.mark.parametrize('directory', [....])` and I've given the exact code that I think should work twice. Does it not work?

Reply:
Hi @ekohl, as @JacobCallahan is a tier two reviewer for this project I will defer to him. See https://github.com/SatelliteQE/robottelo/pull/8400#discussion_r598063542

Thank you

Reply:
And I addressed that by suggesting to split off all the additional work. In https://github.com/SatelliteQE/robottelo/pull/8400#pullrequestreview-618941852 @JacobCallahan was very explicit that you should implement my suggestion. That's why I'm confused.

Reply:
Correct, the directory test should be parameterized with the values you currently have in the directories list. This will remove the need for a loop within the test itself.

Reply:
>  @JacobCallahan was very explicit that you should implement my suggestion. That's why I'm confused.

"the amount of work done before this point would make parameterization **a less than ideal choice**."   that made me think the opposite. Anyway, I have done it now.

Reply:
That is the amount of work in that test, meaning the number of actions taken before it iterates over directories.